src/dataSource.js

```diff
@@ -90,84 +87,201 @@ class DataSource {
+    } else {
+      this.data[row][column] = value;
+    }
+  }
+
+  /**
+   * Get data from the source data set using the physical indexes.
+   *
+   * @private
+   * @param {number} row Physical row index.
+   * @param {string|number|Function} column Physical column index / property / function.
+   * @param {Array|object} dataRow A representation of a data row.
+   * @returns {*} Value at the provided coordinates.
+   */
+  getAtPhysicalCell(row, column, dataRow) {
+    let result = null;
+
+    if (dataRow) {
+      if (typeof column === 'string') {
+        result = getProperty(dataRow, column);
+
+      } else if (typeof column === 'function') {
+        result = column(dataRow);
 
       } else {
-        result = dataRow[prop];
+        result = dataRow[column];
+      }
+    }
+
+    if (this.hot.hasHook('modifySourceData')) {
+      const valueHolder = createObjectPropListener(result);
+
+      this.hot.runHooks('modifySourceData', row, this.colToProp(column), valueHolder, 'get');
+
+      if (valueHolder.isTouched()) {
+        result = valueHolder.value;
       }
     }
 
     return result;
   }
 
+  /**
+   * Returns a single value from the data.
+   *
+   * @param {number} row Physical row index.
+   * @param {number} column Visual column index.
+   * @returns {*}
+   */
+  getAtCell(row, column) {
+    let modifyRowData = null;
+
+    if (this.hot.hasHook('modifyRowData')) {
+      modifyRowData = this.hot.runHooks('modifyRowData', row);
+    }
+
+    const dataRow = modifyRowData !== null && isNaN(modifyRowData) ? modifyRowData : this.data[row];
+
+    return this.getAtPhysicalCell(row, this.colToProp(column), dataRow);
+  }
+
   /**
    * Returns source data by passed range.
    *
-   * @param {object} start Object with physical `row` and `col` keys (or visual column index, if data type is an array of objects).
-   * @param {object} end Object with physical `row` and `col` keys (or visual column index, if data type is an array of objects).
+   * @param {object} [start] Object with physical `row` and `col` keys (or visual column index, if data type is an array of objects).
+   * @param {object} [end] Object with physical `row` and `col` keys (or visual column index, if data type is an array of objects).
    * @param {boolean} [toArray=false] If `true` return source data as an array of arrays even when source data was provided
    *                                  in another format.
    * @returns {Array}
    */
-  getByRange(start, end, toArray = false) {
-    const startRow = Math.min(start.row, end.row);
-    const startCol = Math.min(start.col, end.col);
-    const endRow = Math.max(start.row, end.row);
-    const endCol = Math.max(start.col, end.col);
+  getByRange(start = null, end = null, toArray = false) {
+    let getAllProps = false;
+    let startRow = null;
+    let startCol = null;
+    let endRow = null;
+    let endCol = null;
+
+    if (start === null && end === null) {
```

Comment:
One more thing. I think that if any of them is null then we can't go with `else`
```suggestion
    if (start === null || end === null) {
```

`startRow = Math.min(start.row, end.row)` in `else` block will throw an error if any of them is `null` or `undefined`